src/cryptsetup/cryptsetup.c

```diff
@@ -1136,6 +1137,10 @@ static int attach_luks_or_plain_or_bitlk_by_tpm2(
 
                         assert(!event);
 
+                        if (is_efi_boot() && !efi_has_tpm2())
+                                return log_notice_errno(SYNTHETIC_ERRNO(EAGAIN),
+                                                        "No TPM2 hardware discovered and EFI bios indicates no support for it either, assuming TPM2-less system, falling back to traditional unocking.");
```

Comment:
Is it worth giving an hint on how to configure manually in this case? If we get to this point, it's because somewhere (crypttab?) using the tpm was configured, right?

Reply:
(for the manually added TPM case)

Reply:
honestly i don't know if we can support this at all sanely. at least i don't really know how that would work, because PCRs are all rubbish.

I also doubt ther are any real systems like that out there. 

or to say this differently: once someone shows up with an EFI system with a TPM2 but no firmware support for it, we can revisit the issue, but until then the log message we show should be explanatory enough, no?

Reply:
makes sense, thanks